homeassistant/components/shelly/event.py

```diff
@@ -95,6 +103,21 @@ async def async_setup_entry(
                 async_remove_shelly_entity(hass, EVENT_DOMAIN, unique_id)
             else:
                 entities.append(ShellyRpcEvent(coordinator, key, RPC_EVENT))
+
+        script_instances = get_rpc_key_instances(
+            coordinator.device.status, SCRIPT_EVENT.key
+        )
+        for script in script_instances:
+            event_types = await get_rpc_script_event_types(
+                coordinator.device, int(script.split(":")[-1])
+            )
+            if not event_types:
+                continue
+
+            entities.append(
+                ShellyRpcEvent(coordinator, script, SCRIPT_EVENT, event_types)
```

Comment:
I would suggestcreating a separate class for these entities e.g. `ShellyRpcScriptEvent(ShellyRpcEvent)`

Reply:
Should the current class `ShellyRpcEvent` be renamed to `ShellyRpcInputEvent` to better reflect what that class does?

Reply:
I think `ShellyRpcEvent` is our base class for an event platform and we don't need to change the name.